cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -669,6 +674,30 @@ func (csr *ClusterStateRegistry) GetUnregisteredNodes() []UnregisteredNode {
 	return result
 }
 
+func (csr *ClusterStateRegistry) updateCloudProviderDeletedNodes(deletedNodes []*apiv1.Node) {
+	result := make(map[string]*apiv1.Node, len(deletedNodes)+len(csr.deletedNodes))
+	for _, deleted := range deletedNodes {
+		if prev, found := csr.deletedNodes[deleted.Name]; found {
+			result[deleted.Name] = prev
```

Comment:
Why does this carry over stale `Node` objects from previous iterations? Come to think of it, do we actually need to store anything beyond deleted node names? (We do for taint-based check, but this code is invoked later.)

Reply:
The logic here is mostly copied from a similar implementation.  It isn't necessary to store more than the node name.  I can update everything to use a map of the node name.  Is there a value that makes sense to store?  My initial instinct is to store an epoch for the value of the map.

Reply:
I'd have a value only if it is going to be used for something. Otherwise `struct{}` or `bool` should suffice.

Reply:
I will update the usage to match that suggestion of using the empty struct.